confgenerator/logging.go

```diff
@@ -45,7 +45,7 @@ func setLogNameComponents(ctx context.Context, tag, logName, receiverType string
 }
 
 // stackdriverOutputComponent generates a component that outputs logs matching the regex `match` using `userAgent`.
-func stackdriverOutputComponent(match, userAgent string, storageLimitSize string) fluentbit.Component {
+func stackdriverOutputComponent(match, userAgent string, storageLimitSize string, labels string) fluentbit.Component {
```

Comment:
Nit: `labels` is logically a `map[string]string`, not a `string`, so I'd rather have the parameter be encoding-agnostic and let the function flatten everything to a string internally instead.

While we're at it, can you update the function header to briefly document `storageLimitSize` and `labels`?

Reply:
After debugging bunch of `Fluent-bit + Windows` test failures I realized I couldn't have 3 strackdriver (1 for customer logs, 1 for fluent-bit logs, 1 for health logs) outputs in the config (this seems to be a fluent-bit bug in windows related to threading, synced with @braydonk ).

I opted instead to use a `Modify` filter to add the labels. Some details on the options on adding labels to `ops-agent-health` logs:
- `Output Stackdriver Plugin` : 
    - The only way to add labels in the output plugin without breaking Windows would be to add them for both `ops-agent-health` and `ops-agent-fluent-bit` logs.
    - Adding labels in the output plugin is probably faster that using a `modify` filter.
- `Modify Filter` : 
    - To avoid changing in any way the `ops-agent-fluent-bit` logs, I opted to add all labels from a modify filter.
    - The filter was already being used for setting `severity` and `sourceLocation`.

Reply:
Update : Rewrote some details of last comment as per changes of last commits.

Reply:
Synced offline with @jefferbrecht , @rafaelwestphal , using `Ops Agent - Modify Fields` for the labels.